AllowanceManagement/Repositories/EmployeeRepository.cs

```diff
@@ -1,16 +1,27 @@
 ﻿using AllowanceManagement.Data;
 using AllowanceManagement.Models;
 using AllowanceManagement.Repositories.IRepositories;
+using Microsoft.EntityFrameworkCore;
 
 namespace AllowanceManagement.Repositories
 {
     public class EmployeeRepository : Repository<Employee>, IEmployeeRepository
     {
         public EmployeeRepository(ApplicationDbContext context) : base(context) { }
         public ApplicationDbContext ApplicationDbContext => Context as ApplicationDbContext;
-        public void Update()
-        {
 
+        public IEnumerable<Employee> GetAllEmployeesWithRanksAndCategories()
+        {
+            return Context.Set<Employee>()
+                 .Include(e => e.RankAmount)
+                 .Include(e => e.CategoryPercentage)
+                 .ToList();
```

Comment:
Auto edw ""public ApplicationDbContext ApplicationDbContext => Context as ApplicationDbContext;""
yparxei gia na boreis na kaneis to parakatw.
return ApplicationDbContext.Employees
     .Include(e => e.RankAmount)
     .Include(e => e.CategoryPercentage)
     .ToList();
     
An thes na to kaneis opws to exeis tote apla svhse auto ""public ApplicationDbContext ApplicationDbContext => Context as ApplicationDbContext;""